toolkit/tools/licensecheck/licensecheck.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// A tool for validating the license files of RPM packages in a set of directories.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/exe"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/pkg/licensecheck"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app = kingpin.New("licensecheck", "A tool for validating the license files of RPM packages.")
+
+	rpmDirs       = app.Flag("rpm-dirs", "Directories to recursively scan for RPMs to validate").Required().ExistingDirs()
+	nameFile      = app.Flag("name-file", "File containing license names to check for.").Required().ExistingFile()
+	exceptionFile = app.Flag("exception-file", "File containing license exceptions.").ExistingFile()
+	pedantic      = app.Flag("pedantic", "Enable pedantic mode, warnings are errors.").Bool()
+
+	buildDirPath = app.Flag("build-dir", "Directory to store temporary files.").Required().String()
+	distTag      = app.Flag("dist-tag", "The distribution tag.").Required().String()
+	workerTar    = app.Flag("worker-tar", "Full path to worker_chroot.tar.gz.").Required().ExistingFile()
+
+	logFlags   = exe.SetupLogFlags(app)
+	resultFile = app.Flag("results-file", "The file to store the search result.").Default("").String()
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitBestEffort(logFlags)
+
+	// Run scans on each directory
+	totalResults := []licensecheck.LicenseCheckResult{}
+	totalFailedPackages := 0
+	totalWarningPackages := 0
+	for _, rpmDir := range *rpmDirs {
+		errorResults, warningResults, err := validateRpmDir(*buildDirPath, *workerTar, rpmDir, *nameFile, *exceptionFile, *distTag, *pedantic)
+		if err != nil {
+			logger.Log.Fatalf("Failed to search RPM directory. Error: %v", err)
+		}
+		totalFailedPackages += len(errorResults)
+		totalWarningPackages += len(warningResults)
+		totalResults = append(totalResults, errorResults...)
+		totalResults = append(totalResults, warningResults...)
+	}
+
+	// Save results
+	if *resultFile != "" {
+		logger.Log.Infof("Writing results to file: %s", *resultFile)
+		err := licensecheck.SaveLicenseCheckResults(*resultFile, totalResults)
+		if err != nil {
+			logger.Log.Fatalf("Failed to write results to file. Error: %v", err)
+		}
+	}
+
+	// Print summary
+	if totalFailedPackages > 0 {
+		printExplanation()
+		logger.Log.Errorf("Found (%d) packages with license errors.", totalFailedPackages)
+		logger.Log.Warnf("Found (%d) packages with non-fatal license issues.", totalWarningPackages)
+		os.Exit(1)
+	} else if totalWarningPackages > 0 {
+		printExplanation()
+		logger.Log.Warnf("Found (%d) packages with non-fatal license issues", totalWarningPackages)
+	} else {
+		logger.Log.Infof("No license issues found")
+	}
```

Comment:
Instead of a comment, why not:
```suggestion
	results, failedPackages, warningPackages, err := scanDirectories(...)
	
	if *resultFile != "" {
		logger.Log.Infof("Writing results to file: %s", *resultFile)
		err := licensecheck.SaveLicenseCheckResults(*resultFile, totalResults)
		if err != nil {
			logger.Log.Fatalf("Failed to write results to file. Error: %v", err)
		}
	}

	printSummary(failedPackages, failedWarnings)
```

The results saving part seems pretty clear, since we have both the log saying what we're doing as well as we only call one function with a very clear name.

Reply:
Done